nativelib/src/main/scala/scala/scalanative/native/time.scala

```diff
@@ -0,0 +1,27 @@
+package scala.scalanative
+package native
+
+@extern
+object time {
+
+  type time_t = CLong
+
+  @struct
+  class timespec(
+      val tv_sec: time_t = 0L, /* seconds */
+      val tv_nsec: CLong = 0L /* nanoseconds */
+  )
+
+  def clock_getres(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt  = extern
+  def clock_gettime(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt = extern
+  def clock_settime(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt = extern
```

Comment:
Once https://github.com/scala-native/scala-native/issues/58 lands we'll be able to implement subsets of APIs in platform-dependent manner (e.g. cross project that has two separate implementations for windows and unix-like systems). For now we should avoid using any platform-specific APIs as that would make it hard to get rid of those dependencies later on.